src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs

```diff
@@ -15,25 +22,46 @@ public class HostWarmupMiddleware
         private readonly IScriptWebHostEnvironment _webHostEnvironment;
         private readonly IEnvironment _environment;
         private readonly IScriptHostManager _hostManager;
+        private readonly ILogger _logger;
 
-        public HostWarmupMiddleware(RequestDelegate next, IScriptWebHostEnvironment webHostEnvironment, IEnvironment environment, IScriptHostManager hostManager)
+        public HostWarmupMiddleware(RequestDelegate next, IScriptWebHostEnvironment webHostEnvironment, IEnvironment environment, IScriptHostManager hostManager, ILogger<HostWarmupMiddleware> logger)
         {
             _next = next;
             _webHostEnvironment = webHostEnvironment;
             _environment = environment;
             _hostManager = hostManager;
+            _logger = logger;
         }
 
         public async Task Invoke(HttpContext httpContext)
         {
             if (IsWarmUpRequest(httpContext.Request, _webHostEnvironment, _environment))
             {
+                PreJitPrepare();
+
                 await WarmUp(httpContext.Request);
             }
 
             await _next.Invoke(httpContext);
         }
 
+        private void PreJitPrepare()
+        {
+            // This is to PreJIT all methods captured in coldstart.jittrace file to improve cold start time
+            var path = Path.Combine(Path.GetDirectoryName(new Uri(typeof(HostWarmupMiddleware).Assembly.CodeBase).LocalPath), WarmUpConstants.PreJitFolderName, WarmUpConstants.JitTraceFileName);
+
+            var file = new FileInfo(path);
+
+            if (file.Exists)
+            {
+                JitTraceRuntime.Prepare(file, out int successfulPrepares, out int failedPrepares);
+
+                // We will need to monitor failed vs success prepares and if the failures increase, it means code paths have diverged or there have been updates on dotnet core side.
+                // When this happens, we will need to regenerate the coldstart.jittrace file.
+                _logger.LogInformation($"PreJIT Successful prepares: {successfulPrepares}, Failed prepares: {failedPrepares}");
```

Comment:
Just one nit on this -- can you add an `EventId` -- something like `new EventId(100, "PreJITResults")`? That will allow us to more easily search for the EventName "PreJITResults" in Kusto.

` _logger.LogInformation(new EventId(100, "PreJIT"), $"PreJIT Successful prepares: {successfulPrepares}, Failed prepares: {failedPrepares}");`